[Zi Yang] iP

zylee348 opened the pull request:
# Dukie
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Dukie frees your mind of having to remember things you need to do. It's,


- text-based
- easy to learn
-  ~~FAST~~ **_SUPER FAST_** to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
If you Java programmer, you can use it to practice Java too. Here's the main method:


```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

zylee348 committed:
Level - 1

zylee348 committed:
Level - 2

zylee348 committed:
Level-3

zylee348 committed:
Level-4

zylee348 committed:
Level-5

zylee348 committed:
Level-6

zylee348 committed:
Level-6

zylee348 committed:
branch-level-7

zylee348 committed:
branch-level-7

zylee348 committed:
branch-level-8

zylee348 committed:
A-moreOOP

zylee348 committed:
A-moreOOP

zylee348 committed:
A-Packages

zylee348 committed:
A-Gradle

zylee348 committed:
A-JUnit

zylee348 committed:
A-Jar

zylee348 committed:
A-JavaDocs

zylee348 committed:
Revert "A-JavaDocs"

This reverts commit 7a8743e40379a8ca0d838fb8665ca21d0d40fcc5.

zylee348 committed:
Revert "Revert "A-JavaDocs""

This reverts commit 4286cd583b509bd156c8ce57c3443b42beed7965.

zylee348 committed:
Revert "Revert "Revert "A-JavaDocs"""

This reverts commit 685d1d4f14425cb59cc1bf3abb85d20c960aa555.

zylee348 committed:
Revert "Revert "Revert "Revert "A-JavaDocs""""

This reverts commit e1332f4db7a437ba7c03985a7424526f1571f946.

zylee348 committed:
branch-A-JavaDoc

zylee348 committed:
branch-Level-9

zylee348 committed:
Merge tag 'branch-A-JavaDoc'

zylee348 committed:
Level-9

lpohsien commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+ */
+public class Deadline extends Task {
+
+    private LocalDateTime by;
```
Perhaps a more descriptive name such as `deadline` could be used? Maybe a similar change could be made for the `at` variable in `Event` task.

lpohsien commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,98 @@
+     */
+    public static void parseInput() {
+        Scanner sc = new Scanner(System.in);
+        boolean running = true;
```
Perhaps it would be better if the variable name is renamed ti indicate that it is boolean.

lpohsien commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,53 @@
+        return (isDone ? 1 : 0 ) + " | " + this.description;
+    }
+
+<<<<<<< HEAD
```
Maybe it is better to remove the conflict markers to prevent it from potentially causing problem with git in the future?

lpohsien commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+                String taskString = sc.nextLine();
+                String[] taskStringInArray = taskString.split(" \\| ");
+                String taskType = taskStringInArray[0];
+                if (taskStringInArray.length > 1) {
```
Perhaps you could consider the "Make the app path prominent" technique here by changing the inequality. This might also helps to make the conditions less nested.

lpohsien commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,151 @@
+    public TaskList(ArrayList<Task> taskList) throws DukeException {
+        try {
+            this.taskList = taskList;
+        } catch (Exception e) {
```
I am not sure if the try-catch here is really necessary. From what I understand, if any error did occur (possibly when loading the list), it should be handled by either the `Storage` class or the `Parser` class? Perhaps you might want to change the `Exception` to be more specific so as to avoid catching all exceptions when debugging?

lpohsien reviewed (commented):
Overall, I found the code easy to understand. Most of the coding standard was closely adhered to. Naming for most of the variables are concise yet explanatory and helped me to understand the code more easily. Some minor adjustments could be made to further improve the quality of the code.

zylee348 committed:
branch-Level-10

zylee348 committed:
Flip dialog box

zylee348 committed:
setting up fxml

zylee348 committed:
Added fxml files and Main, MainWindow.

zylee348 committed:
Refactored code to work with GUI.

zylee348 committed:
improving error handling

zylee348 committed:
edit to error messages

zylee348 committed:
Merge tag 'branch-Level-10'

TheSoggy reviewed (commented):
Overall, LGTM! Coding standard was mostly adhered to, with some minor exceptions that you may want to consider changing.

TheSoggy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,87 @@
+     * Parses user input.
+     */
+    public static String parseInput(String string) {
+        //boolean running = true;
```
perhaps you may want to remove commented out code?

TheSoggy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+     * @return arraylist of type task.
+     * @throws IOException if file reading issues arise.
+     */
+    public static ArrayList<Task> loadTask() throws IOException {
```
This method handles a few exceptions and prints out the error to the UI. May I suggest for it to handle the IOException as well so that this method will not throw any exception?

TheSoggy commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,117 @@
+        ArrayList<Task> taskArrayList = TaskList.getTaskList();
+        int counter = 0;
+        int numbering = 1;
+        //getLine();
```
perhaps you may want to remove commented out code?

TheSoggy reviewed (commented):
Overall, LGTM! Coding standard was mostly adhered to, with some minor exceptions that you may want to consider changing.

zylee348 committed:
A-CheckStyle

zylee348 committed:
A-Assertions

zylee348 committed:
parseInput command: removed unrelated comments
getLine command: removed method

Comments and disused methods were distracting and disused. Removed it
to improve code-quality.

Added javadocs for printList method in UI.java

zylee348 committed:
findTasks command: used streams instead

findTasks command was not using streams and was disorganised.

Using streams means more readable code without commenting since
streams have methods which are significantly easier to understand.

Let's,
* update the code to use streams

zylee348 committed:
Merge pull request #1 from zylee348/A-CodeQuality

parseInput command: removed unrelated comments

zylee348 committed:
Merge branch 'master' of https://github.com/zylee348/ip

zylee348 committed:
Merge branch 'master' into A-Streams

zylee348 committed:
Merge branch 'master' into branch-A-Assertions

zylee348 committed:
Merge branch 'master' into branch-A-CodeQuality

zylee348 committed:
Merge pull request #2 from zylee348/branch-A-Assertions

Branch a assertions

zylee348 committed:
Merge branch 'master' of https://github.com/zylee348/ip

zylee348 committed:
Merge branch 'branch-A-Streams'

zylee348 committed:
Removed getLogo method.

getLogo method disused upon GUI usage.

zylee348 committed:
Merge pull request #3 from zylee348/branch-A-CodeQuality

Branch a code quality

zylee348 committed:
Merge branch 'master' of https://github.com/zylee348/ip

zylee348 committed:
Removed more comments

zylee348 committed:
Added message that tasks have been marked and unmarked if previously
done so.

Also added BCD-extension for C-Sort.

zylee348 committed:
Updated README.md in docs, added Ui.png

zylee348 committed:
Last few commits; with jar release

zylee348 committed:
Removed ui.png

zylee348 committed:
Added Ui.png

zylee348 committed:
Add A-CI

zylee348 committed:
Edit CI

zylee348 committed:
Changing permission of gradlew

zylee348 committed:
Edit gradle

zylee348 committed:
Edit Test files

zylee348 committed:
Edit yml file

zylee348 committed:
Edit yml file

zylee348 committed:
Merge pull request #5 from zylee348/branch-A-CI

Branch A-CI
